Dockerfile

```diff
@@ -1,6 +1,28 @@
+# Use a UUID as placeholder value to have a unique string to replace. 
+ARG BASE_URL_PLACEHOLDER=189b303e-37a0-4f6f-8c0a-50333bc3c36e
+
 FROM docker.io/library/node:16.13.2 as build
+
+ARG BASE_URL_PLACEHOLDER
+
+COPY package.json package-lock.json ./
+RUN npm install --frozen-lockfile
+
 COPY ./ ./
```

Comment:
Why these commands are split? What we have benefit here?

Reply:
That's a common optimization to speed up Docker builds. It will cache the result of `npm install` and not execute it unless package.json or package-lock.json have changed.